storage/innobase/dict/dict0stats.cc

```diff
@@ -430,12 +430,15 @@ dict_table_schema_check(
 
 		/* check length for exact match */
 		if (req_schema->columns[i].len != table->cols[j].len) {
-			sql_print_warning("InnoDB: Table %s has"
-					  " length mismatch in the"
-					  " column name %s."
-					  " Please run mariadb-upgrade",
-					  req_schema->table_name_sql,
-					  req_schema->columns[i].name);
+			snprintf(errstr, errstr_sz,
+				 "Table %s has length mismatch in the column "
+				 "name %s. Please either run 'mariadb-upgrade' or fix it "
+				 "manually with: ALTER TABLE %s MODIFY %s;",
+				 req_schema->table_name_sql,
+				 req_schema->columns[i].name,
+				 req_schema->table_name_sql,
+				 req_schema->columns[i].name);
```

Comment:
The `ALTER TABLE` statement is not valid even though it ends in a semicolon. I’d suggest to shorten this message to:
```c++
snprintf(errstr, errstr_sz, "Length mismatch in %s.%s. Run mariadb-upgrade or ALTER TABLE.",
         req_schema->table_name_sql, req_schema->columns[i].name);
```